### ISSUE TITLE
[java] JUnitTestsShouldIncludeAssert false positive with BDDMockito's then()

### ISSUE THREAD

**Issue by krdabrowski** [pmd/pmd#4781]
**Affects PMD Version:** 
7.0.0-rc4

**Rule:**
[JUnitTestsShouldIncludeAssert](https://docs.pmd-code.org/latest/pmd_rules_java_bestpractices.html#junittestsshouldincludeassert)

**Description:**
Rule complaints about missing assertion, even though that BDDMockito's `then()` assertion/verification is present. Probably `TestFrameworksUtil#isProbableAssertCall`: https://github.com/pmd/pmd/blob/a558fd17eb6d8f75a7e5f80c83249d96d0b11043/pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/internal/TestFrameworksUtil.java#L180 should be extended with:
https://javadoc.io/doc/org.mockito/mockito-core/latest/org/mockito/BDDMockito.html#then(T)

**Code Sample demonstrating the issue:**

```java
import static org.mockito.BDDMockito.then;

import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;
import org.mockito.InjectMocks;
import org.mockito.Mock;
import org.mockito.junit.jupiter.MockitoExtension;

@ExtendWith(MockitoExtension.class)
public class FooTest {
    @Mock
    Bar barMock;

    @InjectMocks
    Foo foo;

    @Test
    void test() {
        foo.doTask();

        then(barMock).should().actuallyDoTask();
    }

    class Foo {}
    class Bar {}
}
```

**Expected outcome:**
PMD reports a violation `JUnit tests should include assert() or fail()` at line 21, but that's wrong. That's a false positive - method `then()` is a valid possibility to perform assertions.

**Running PMD through:**
Maven

**Comment by oowekyala** [comment]
Seems to me it's `should` that's doing the assertion?

**Comment by krdabrowski** [comment]
@oowekyala you are right - more precisely it's `then` combined with `should`, `shouldHaveNoMoreInteractions` or `shouldHaveNoInteractions`.
